### PR TITLE
Add support for pluggable npm clients

### PR DESCRIPTION
**Pull request by gigabo**
Allows bootstrap to use `yarn` (for example).

Command-line: `lerna bootstrap --npm-client=yarn`

In lerna.json: `"npmClient": "yarn"`

Relates to https://github.com/lerna/lerna/issues/371.

Pretty small change with the groundwork laid by https://github.com/lerna/lerna/pull/576.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -643,6 +643,24 @@ can be used to opt out of hoisting for certain dependencies.
 $ lerna bootstrap --hoist --nohoist=babel-*
 ```
 
+#### --npm-client [client]
+
+Install external dependencies using `[client] install`.  Must be an executable
+that knows how to install npm dependencies.
```
Comment (doug-wade) [inline README.md]: why only install?

**Comment by wtgtybhertgeghgtwtg** [comment]
Shouldn't it be called `package-manager` instead of `npm-client` or something?

**Comment by gigabo** [comment]
> Shouldn't it be called `package-manager` instead of `npm-client` or something?

It's the client that Lerna uses to connect to the npm registry.

NPM yarn and the like ["third-party registry clients"](http://blog.npmjs.org/post/151660845210/hello-yarn).

File: README.md (hunk 1 above)
Comment (gigabo) [inline README.md]: Because that's the only command it's used for. 😁 

Maybe better to say "Must be a compatible npm registry client"?

**Comment by wtgtybhertgeghgtwtg** [comment]
Does `lerna` specify the `npm` registry?  I think `yarn` tries to pull from its own registry, first.

**Comment by timdp** [comment]
Wasn't there talk of folding monorepo functionality into yarn at one point? That'd make this PR obsolete, right?

**Comment by karlhorky** [comment]
> Wasn't there talk of folding monorepo functionality into yarn at one point?

Yep, here https://github.com/lerna/lerna/issues/371#issuecomment-279339181

**Comment by hzoo** [comment]
@timdp Yep! That requires a rewrite of Lerna into Yarn and would likely take more than a little while. For everyone that wants to continue using Lerna in the meantime there's is and why we continue to work on in the first place

**Comment by timdp** [comment]
Interesting, thanks! :+1:

**Comment by rexxars** [comment]
Want. What's the blocker for getting this merged?

**Comment by neeschit** [comment]
This is awesome!! Thanks a ton!

**Comment by hulkish** [comment]
+1

**Pull request merged**